draft-ietf-quic-transport.md

```diff
@@ -925,18 +925,23 @@ handshake.  The sequence number of the initial connection ID is 0.  If the
 preferred_address transport parameter is sent, the sequence number of the
 supplied connection ID is 1.
 
-Additional connection IDs are communicated to the
-peer using NEW_CONNECTION_ID frames ({{frame-new-connection-id}}).  The
-sequence number on each newly-issued connection ID MUST increase by 1. The
-connection ID randomly selected by the client in the Initial packet and any
-connection ID provided by a Reset packet are not assigned sequence numbers
-unless a server opts to retain them as its initial connection ID.
+Additional connection IDs are communicated to the peer using NEW_CONNECTION_ID
+frames ({{frame-new-connection-id}}).  The sequence number on each newly-issued
+connection ID MUST increase by 1. The connection ID randomly selected by the
+client in the Initial packet and any connection ID provided by a Reset packet
+are not assigned sequence numbers unless a server opts to retain them as its
+initial connection ID.
 
 When an endpoint issues a connection ID, it MUST accept packets that carry this
 connection ID for the duration of the connection or until its peer invalidates
 the connection ID via a RETIRE_CONNECTION_ID frame
 ({{frame-retire-connection-id}}).
 
+Endpoints store received connection IDs for future use.  An endpoint that
+receives excessive connection IDs MAY discard those it cannot store.  An
+endpoint that issues connection IDs cannot expect its peer to store and use all
+issued connection IDs.
```

Comment:
Belated question: when you discard them, it seems you should not send a RETIRE_CONNECTION_ID frame, otherwise you may just get more?

Reply:
The idea is that you can have an uncoordinated negotiation:  I'm willing to issue 64, you only need 8.  You take the 8 you want, let the remaining 56 fall on the floor, and then I'll replace the 8 you kept as you RETIRE them.  If you RETIREd the 56 you didn't need, I'd replace *those*, and we'd get into an endless RETIRE/NEW drop for no purpose.